# 2.3 Thermodynamic properties of R-134a

| T, degC     | P, kPa      |u, kJ/kg  | Phase description|
| ----------- | ----------- |----------|------------------|
|   | 800         | 275      ||
| -10         |        |     | Saturated vapor  |
| 30          | 1200        |     | |
|     | 2000        | 250      | |


In [1]:
from CoolProp import CoolProp as CP
###========== (a)================###
P = 0.8e6 # in Pa
u = 275 # in kJ/kg
fluid = "R134a"
uf = CP.PropsSI("U", "Q", 0, "P", P,fluid)/1e3
ug = CP.PropsSI("U", "Q", 1, "P", P, fluid)/1e3
print("Uf at given pressure: {} kJ/kg".format(round(uf,1)))
print("Ug at given pressure: {} kJ/kg".format(round(ug,1)))
## we are in saturated mixture region; since u_given > uf and u_given < ug!
T = CP.PropsSI("T","P",P,"Q",0,fluid)
print("Phase description: Saturated mixture")
print("Temperature: {}".format(round(T - 273.15,2)),"deg C")

Uf at given pressure: 243.0 kJ/kg
Ug at given pressure: 395.0 kJ/kg
Phase description: Saturated mixture
Temperature: 31.33 deg C


In [2]:
###========== (b)================###
T = -10 + 273.15 ## K
fluid = "R134a"
press1 = CP.PropsSI("P","T",T,"Q",1,fluid)/1e3
ug = CP.PropsSI("U", "Q", 1, "T", T, fluid)/1e3
print("U = Ug at given temperature: {} kJ/kg".format(round(ug,2)))
print("Pressure = {} kPa".format(round(press1,1)))

U = Ug at given temperature: 372.69 kJ/kg
Pressure = 200.6 kPa


In [3]:
###========== (c)================###
T = 30 + 273.15 ## K
P = 1200e3 # Pa
fluid = "R134a"
press_at_given_temp = CP.PropsSI("P","T",T,"Q",0,fluid) /1e3
print("Pressure = {} kPa".format(round(press_at_given_temp,1)))
u= CP.PropsSI("U", "P", P, "T", T, fluid)/1e3
print("U = {} kJ/kg".format(round(u,2)))
uf= CP.PropsSI("U", "Q",0, "T", T, fluid)/1e3
print("Uf = {} kJ/kg".format(round(uf,2)))

Pressure = 770.2 kPa
U = 240.7 kJ/kg
Uf = 241.07 kJ/kg


In [4]:
###========== (d)================###
P = 2000e3 # Pa
fluid = "R134a"
T_sat = CP.PropsSI("T", "P", P,"Q",0,fluid)
print("Temperature = {} C".format(round(T_sat-273.15,1)))
uf= CP.PropsSI("U", "P", P, "Q",0, fluid)/1e3
ug= CP.PropsSI("U", "P", P, "Q",1, fluid)/1e3

print("Uf = {} kJ/kg".format(round(uf,2)))
print("Ug = {} kJ/kg".format(round(ug,2)))
## given U is below u_f; hence compressed liquid
T = CP.PropsSI("T","U",250e3,"P",P,fluid) - 273.15
print("Temperature = {} C".format(round(T,2)))

Temperature = 67.5 C
Uf = 297.98 kJ/kg
Ug = 409.7 kJ/kg
Temperature = 36.92 C


## completed table:
| T, degC     | P, kPa      |u, kJ/kg  | Phase description|
| ----------- | ----------- |----------|------------------|
|     **31.33**   | 800         | 275      |**Saturated liq-vapour mixture** |
| -10         |    **200.6**    | **372.687**    | Saturated vapor  |
| 30          | 1200        |  **240.7**   | **Compressed liquid**|
|   **36.92**  | 2000        | 250      |**Compressed Liquid** |
